### PR TITLE
SETTINGS: Add use_spi, spi_motor_drivers, lc_daughter_card

### PR DESCRIPTION
**Pull request by StephanCoetzee**
@Jeff-Ciesielski

### CODE REVIEW

**Comment by StephanCoetzee** [comment]
Tested

File: main.c [hunk 1]
```diff
@@ -115,10 +117,10 @@ int main(void)
     systick_init();  // Init systick and systick callbacks
 
     /* Initialize digital potentiometers */
-#ifdef SPI_STEPPER_DRIVER
-    ad5121_init(AD5121_GAIN);
-    ad5121_init(AD5121_CAL);
-#endif
+    if (settings.use_spi) {
```
Comment (Jeff-Ciesielski) [inline main.c]: These should probably be wrapped in a check for the load cell daughter card.

File: stepper.c [hunk 2]
```diff
@@ -683,13 +683,10 @@ void keyme_init()
   ESTOP_PORT &= ~(ESTOP_BIT);           //estop input normal-low
 
   // Microstepping
-
-  #ifdef SPI_STEPPER_DRIVER
-    /* TODO: Initialize motor drivers with tables here */
-  #else
+  if (settings.spi_motor_drivers == 0) {
```
Comment (Jeff-Ciesielski) [inline stepper.c]: Seems a little odd that the motors have bits and pieces of init code laying around different places in the codebase.  We should probably think about writing a motor driver abstraction layer (or expanding motor_drv) and consolidating this stuff.

**Review by Jeff-Ciesielski — COMMENTED** [review]
The abstraction comment is a future suggestion. Other than the check for lc_daughter_card, this looks pretty strait forward.

**Comment by StephanCoetzee** [comment]
Updated

**Comment by Jeff-Ciesielski** [comment]
LGTM after rebase + Test

**Comment by StephanCoetzee** [comment]
Rebased. Tested by enabling / disabling `use spi` and `spi motor drivers` and testing the motion of the motors accordingly.

**Comment by StephanCoetzee** [comment]
Ready to merge @Jeff-Ciesielski

**Comment by Jeff-Ciesielski** [comment]
LGTM

**Pull request merged**